ikasaneip/component/endpoint/mongo-endpoint-test/src/main/java/org/ikasan/component/endpoint/mongo/test/EmbeddedMongo.java

```diff
@@ -1,278 +1,304 @@
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL 
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE 
+ * USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ * =============================================================================
+ */
+package org.ikasan.component.endpoint.mongo.test;
+
+import com.mongodb.MongoClient;
+import com.mongodb.ServerAddress;
+import de.flapdoodle.embed.mongo.Command;
+import de.flapdoodle.embed.mongo.MongodExecutable;
+import de.flapdoodle.embed.mongo.MongodProcess;
+import de.flapdoodle.embed.mongo.MongodStarter;
+import de.flapdoodle.embed.mongo.config.*;
+import de.flapdoodle.embed.mongo.distribution.Feature;
+import de.flapdoodle.embed.mongo.distribution.Version;
+import de.flapdoodle.embed.mongo.distribution.Versions;
+import de.flapdoodle.embed.process.config.IRuntimeConfig;
+import de.flapdoodle.embed.process.config.store.HttpProxyFactory;
+import de.flapdoodle.embed.process.distribution.GenericVersion;
+import de.flapdoodle.embed.process.io.directories.IDirectory;
+import de.flapdoodle.embed.process.runtime.Network;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.UnknownHostException;
+
+/**
+ * Allows the port and download location of the mongo distribution to be configured Also ensures that only a single
+ * embedded mongo instance is started for all tests
+ * 
+ * @author Kieron Edwards
+ */
+public class EmbeddedMongo
+{
+    private final static Logger logger = LoggerFactory.getLogger(EmbeddedMongo.class);
+
+    private MongodExecutable mongodExecutable;
+
+    private static EmbeddedMongo staticReference;
+
+    private MongoClient mongoClient;
+
+    private final EmbeddedMongoConfiguration configuration;
+
+
+    /**
+     * @deprecated - use the configuration object constructor or the empty constructor instead
+     */
+    @Deprecated
+    public EmbeddedMongo(int port)
+    {
+        super();
+        configuration = new EmbeddedMongoConfiguration();
+        configuration.setPort(port);
+        overrideConfigurationWithEnvironmentVariables();
+        overrideConfigurationWithSystemProperties();
+    }
+
+    /**
+     * This will automatically allocated a free port to the mongo instance
+     */
+    public EmbeddedMongo()
+    {
+        configuration = new EmbeddedMongoConfiguration();
+        overrideConfigurationWithEnvironmentVariables();
+        overrideConfigurationWithSystemProperties();
```

Comment:
Wont this apply the environment variables then apply the system properties, with the latter over writing the former? Is that the correct way around?

Reply:
Yes, I intended to make the latter (system properties) override the former (environment variables).
Priority should be given to the system properties as they are provided by by developer or pipeline via command line.